dynamicio/handlers/file.py

```diff
@@ -0,0 +1,83 @@
+"""File handlers for dynamicio."""
+from copy import deepcopy
+from pathlib import Path
+from threading import Lock
+from typing import Any, Dict
+
+import pandas as pd
+from pydantic import Field
+
+from dynamicio import utils
+from dynamicio.base import BaseResource
+from dynamicio.inject import check_injections, inject
+
+hdf_lock = Lock()
+
+
+class BaseFileResource(BaseResource):
+    """Base class for file resources."""
+
+    path: Path
+    kwargs: Dict[str, Any] = {}
+
+    def _check_injections(self) -> None:
+        """Check that all injections have been completed."""
+        check_injections(str(self.path))
+
+    def inject(self, **kwargs) -> "BaseFileResource":
+        """Inject variables into path. Not in place."""
+        new = deepcopy(self)
+        new.path = inject(str(new.path), **kwargs)  # type: ignore
+        return new
+
+
+class HdfFileResource(BaseFileResource):
+    """HDF file resource."""
+
+    pickle_protocol: int = Field(4, ge=0, le=5)  # Default covers python 3.4+
+
+    def _resource_read(self) -> pd.DataFrame:
+        """Read from HDF file."""
+        with hdf_lock:
```

Comment:
Why is a lock necessary?

Reply:
No Idea, just refactoring from the previous version.